### ISSUE TITLE
Single window mode

### ISSUE THREAD

**Issue by Pintaio** [HaikuArchives/ArtPaint#482]
Please, consider adding a single window mode to Artpaint. I think a single window layout is more comfortable
Else, you can do something similar to what was done with Gimp, so you can choose between single or multi window mode.

**Comment by PeteCA** [comment]
i feel the need to inject my (personal) diametrically opposite opinion here!
I normally *hate* the constraints of single window mode, at least for any
reasonably complex app.  Haiku's ease of easily hiding and restoring windows
makes personally configuring a display very friendly (not so easy  in otherOSs).
[see my MusicWeaver for an extreme example!  There's no way that could be
done single-window...]

**Comment by dsizzle** [comment]
well, from a purely technical standpoint, I have no idea how a single-window mode might be implemented - afaik Haiku doesn't have any sort of dockable Window built-in.  

Having said that, I believe that the Haiku way to do a reasonable compromise would be via Stack & Tile. That way you could in theory connect all your windows together as you like.  I don't know how to use it yet nor how robust it is, but it is a possibility.

**Comment by humdingerb** [comment]
Stacking via API is already possible, Koder does it. AFAIK tiling, which may be more useful for ArtPaint to tile e.g. the Tools and Layers window to the main window, isn't yet.

Even then, tiling windows resizes them to fit together, which isn't really what we want. I filed an enhancement for that area a while back:[Add 'glue' mode to S&T to keep window's original size](https://github.com/ticket/17696)](https://dev.haiku-os.org/ticket/17696)

Also, what to do when you have several image windows open? Always un-tile and tile them to the active image window?

Maybe ArtPaint could (optionall) stack all image windows like Koder. Then those are at least not scattered around and people could (manually or eventually automatically) stack and tile the tools, layers etc. windows around it.

**Comment by dsizzle** [comment]
> Maybe ArtPaint could (optionall) stack all image windows like Koder. 

that’s not a bad idea. The problem I have with Koder though is the paths get cut off when you have several tabs stacked. Maybe default to just the file names.

maybe windows should also be added to bottom of the Window menu so they’re easy to see/access?

**Comment by humdingerb** [comment]
> Maybe default to just the file names.

Yes. Maybe, if a file with the same name but different path is opened, it gets the relevant path added. For example:

Already open: `/boot/home/data/graphics/photos/DaleInArmor.jpg` shown as **DaleInArmor.jpg**
Open: `/boot/home/data/graphics/old-photos/DaleInArmor.jpg` shown as **old-photos/DaleInArmor.jpg**

Trancated in the middle if it gets too long still.
Also, this automatic stacking could be limited to 5 or 6 (or whatever) number of images before opening a new stack-o-pics.

> maybe windows should also be added to bottom of the Window menu so they’re easy to see/access?

Good idea, too. 👍
